packages/snaps-simulation/src/types.ts

```diff
@@ -405,6 +410,16 @@ export type Snap = {
    */
   onHomePage(): Promise<SnapResponseWithInterface>;
 
+  /**
+   * Send a keyring to the Snap.
+   *
+   * @param keyringRequest - Keyring request options.
+   * @returns The response.
+   */
+  onKeyringRequest(
+    keyringRequest: Partial<KeyringOptions>,
```

Comment:
Not convinced this should be partial. If we use the non-partial type, method will be required, but that seems fine

Reply:
Updated.